sound/soc/sof/core.c

```diff
@@ -230,9 +231,17 @@ static int sof_probe_continue(struct snd_sof_dev *sdev)
 	 * Register audio client.
 	 * This can fail but errors cannot be propagated.
 	 */
-	sdev->sof_audio = sof_client_dev_register(sdev, "sof-audio");
-	if (!sdev->sof_audio)
-		dev_warn(sdev->dev, "sof-audio client failed to register\n");
+	sdev->sof_ssp_audio = sof_client_dev_register(sdev, "sof-ssp-audio");
+	if (!sdev->sof_ssp_audio)
+		dev_warn(sdev->dev, "sof-ssp-audio client failed to register\n");
+
+	/*
+	 * Register HDMI audio client.
+	 * This can fail but errors cannot be propagated.
+	 */
+	sdev->sof_hdmi_audio = sof_client_dev_register(sdev, "sof-hdmi-audio");
+	if (!sdev->sof_hdmi_audio)
+		dev_warn(sdev->dev, "sof-hdmi-audio client failed to register\n");
 
 	if (plat_data->sof_probe_complete)
```

Comment:
I don't really agree with this change. core.c should be a generic file. Here we need to have a list of clients and iterate through them for registration.

Reply:
fixed.